transport-native-unix-common/src/main/c/netty_unix_errors.c

```diff
@@ -49,8 +51,24 @@ void netty_unix_errors_throwRuntimeExceptionErrorNo(JNIEnv* env, char* message,
 
 void netty_unix_errors_throwChannelExceptionErrorNo(JNIEnv* env, char* message, int errorNumber) {
     char* allocatedMessage = exceptionMessage(message, errorNumber);
-    (*env)->ThrowNew(env, channelExceptionClass, allocatedMessage);
+    jstring errorMessage = (*env)->NewStringUTF(env, allocatedMessage);
     free(allocatedMessage);
+    allocatedMessage = NULL;
+    if (errorMessage == NULL) {
+      return;
```

Comment:
why not keep `(*env)->ThrowNew(env, channelExceptionClass, allocatedMessage);` here if alloc of NewStringUTF fails?

Reply:
obviously freeing of allocatedMessage is going to cause problems..

Reply:
Because it fails when there is an OOME in progress.  Throwing another exception would either fail with another OOME, or replace the OOME.  I don't know how the second case can occur (NewStringUTF docs are unclear if NULL implies exception).  I could add a `assert((*env)->ExceptionOccurred() !=NULL);`

Reply:
I think it's fine as is.